ZeroWasteFriendly/Scenes/EcoMap/EcoMapInteractor.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  EcoMapInteractor.swift
+//  ZeroWasteFriendly
+//
+//  Created by Marek Skrzelowski on 20/02/2020.
+//  Copyright © 2020 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+protocol EcoMapInteractor {
+    func getUserLocation()
+    func getPlaces()
+}
+
+final class EcoMapInteractorImpl {
+
+    private let presenter: EcoMapPresenter
+    private let worker: EcoMapWorker
+    private let router: EcoMapRouter
+
+    init(presenter: EcoMapPresenter,
+         worker: EcoMapWorker,
+         router: EcoMapRouter) {
+        self.presenter = presenter
+        self.worker = worker
+        self.router = router
+    }
+}
+
+extension EcoMapInteractorImpl: EcoMapInteractor {
+
+    func getUserLocation() {
+        presenter.toggleSpinner(true)
+        worker.fetchUserLocation { [weak self] result in
```

Comment:
I think I'd remove these [weak self] so self can stay iin memory during requests so we can be sure that we dismiss the spinner